src/Report/ReportInterface.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Report;
+
+use Symfony\Component\Stopwatch\Stopwatch;
+
+interface ReportInterface
+{
+    /**
+     * @param int $showAppliedFixers
+     */
+    public function setShowAppliedFixers($showAppliedFixers);
+
+    /**
+     * @param bool $needDiff
+     */
+    public function setShowDiff($needDiff);
+
+    /**
+     * @param bool $dryRun
+     */
+    public function setDryRun($dryRun);
+
+    /**
+     * @param bool $decoratedOutput
+     */
+    public function setDecoratedOutput($decoratedOutput);
+
+    /**
+     * Sets the Stopwatch.
+     *
+     * @param Stopwatch $stopwatch
+     */
+    public function setStopwatch(Stopwatch $stopwatch);
```

Comment:
All the setters should not be part of interface.
For example: If one want to build hist own report that doesn't care about Stopwatch he should not be forced to create such method.

Take a look what is the reporter for - to create report. User of Reporter doesn't care how the report was created, if for example dry-run or stopwatch was used or not.

Please, remove the methods from interface. I would even go a bit farther and don't have them at abstract level as well - instead of that pass things that really matters for concrete reporters in constructor.